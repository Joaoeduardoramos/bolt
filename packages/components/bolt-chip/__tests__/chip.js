const { render, renderString } = require('@bolt/twig-renderer');

describe('<bolt-chip> Component', async () => {
  test('basic usage with attributes', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      attributes: {
        'data-attr': 'some attribute',
      },
      text: 'This is a chip',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('with link', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      text: 'This is a chip with link',
      url: '!#',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
