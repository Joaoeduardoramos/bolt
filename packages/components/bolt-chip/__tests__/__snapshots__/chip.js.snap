// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chip basic usage 1`] = `
<bolt-chip>
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-small"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Hello World
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip chip size: medium 1`] = `
<bolt-chip size="medium">
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-medium"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Chip size
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip chip size: small 1`] = `
<bolt-chip size="small">
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-small"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Chip size
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip chip size: xsmall 1`] = `
<bolt-chip size="xsmall">
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-xsmall"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Chip size
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip icon usage 1`] = `
<bolt-chip>
  <replace-with-grandchildren>
    <span class="c-bolt-chip c-bolt-chip--size-small"
          is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__icon">
        <bolt-icon slot="before"
                   name="check"
        >
        </bolt-icon>
      </replace-with-children>
      <replace-with-children class="c-bolt-chip__text">
        Icon before
      </replace-with-children>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`chip url usage 1`] = `
<bolt-chip url="https://pega.com"
           target="_blank"
>
  <replace-with-grandchildren>
    <a href="https://pega.com"
       target="_blank"
       class="c-bolt-chip c-bolt-chip--link c-bolt-chip--size-small"
       is="shadow-root"
    >
      <replace-with-children class="c-bolt-chip__text">
        Has URL
      </replace-with-children>
    </a>
  </replace-with-grandchildren>
</bolt-chip>
`;
